packages/sync-actions/test/product-types-sync-attribute-hints.spec.js

```diff
@@ -0,0 +1,492 @@
+                  newValue: attributeEnumDraftItem.next,
+                },
+              ])
+            )
+          })
+        })
+        describe('when is localized', () => {
+          beforeEach(() => {
+            attributeEnumDraftItem = createAttributeEnumDraftItem({
+              next: {
+                key: 'next-key',
+                label: 'next-label',
+              },
+              hint: {
+                isLocalized: true,
+                attributeName: 'attribute-name',
+              },
+            })
+            updateActions = sync.buildActions(
+              {},
+              {},
+              {
+                nestedValuesChanges: {
+                  attributeEnumValues: [attributeEnumDraftItem],
+                },
+              }
+            )
+          })
+          it('should match snapshot', () => {
+            expect(updateActions).toMatchSnapshot()
+          })
+          it('should generate `changeEnumKey` update-action', () => {
+            expect(updateActions).toEqual(
+              expect.arrayContaining([
+                {
+                  action: 'changeEnumKey',
+                  attributeName: attributeEnumDraftItem.hint.attributeName,
+                  key: 'enum-key',
+                  newKey: 'next-key',
+                },
+              ])
+            )
+          })
+          it('should generate `changeLocalizedEnumValueLabel` update-action', () => {
+            expect(updateActions).toEqual(
+              expect.arrayContaining([
+                {
+                  action: 'changeLocalizedEnumValueLabel',
+                  attributeName: attributeEnumDraftItem.hint.attributeName,
+                  newValue: attributeEnumDraftItem.next,
+                },
+              ])
+            )
+          })
+        })
+
+        describe('when removing, adding, and editing (in a single batch of actions)', () => {
+          let attributeEnumDraftItemToBeRemoved1
+          let attributeEnumDraftItemToBeRemoved2
+          let attributeEnumDraftItemToBeChanged
+          let attributeEnumDraftItemToBeAdded
+          beforeEach(() => {
+            attributeEnumDraftItemToBeRemoved1 = createAttributeEnumDraftItem({
+              previous: { key: 'enum-key-1', label: 'enum-label-1' },
+              next: undefined,
+              hint: {
+                attributeName: 'attribute-enum-with-2-enum-values-to-remove',
+                isLocalized: false,
+              },
+            })
+            attributeEnumDraftItemToBeRemoved2 = createAttributeEnumDraftItem({
+              previous: { key: 'enum-key-2', label: 'enum-label-2' },
+              next: undefined,
+              hint: {
+                attributeName: 'attribute-enum-with-2-enum-values-to-remove',
+                isLocalized: false,
+              },
+            })
+            attributeEnumDraftItemToBeChanged = createAttributeEnumDraftItem({
```

Comment:
Im confused here. How can this become a `changePlainEnumValueLabel` when there is no `previous`?

Reply:
Makes sense. These `previous` and `next` is something I'd have to figure out myself, as a user, then? They won't be generated?

Reply:
Got it. Yeah I'd guess just some communication problems. But its starting to get clear for me now. 

So, in essence, we can say that proposed PR will calculate all `base actions` and let the user calculate and provide it for `attributes`/`enums` themselves? Do I understand correctly then?

Reply:
exactly. if they want their nested values calculated, they either...

1. calculate it themselves, which they have to anyways since I intended to have product-types removed entirely
2. use hints to help the sync-actions calculate appropriate actions

cc @junajan